hoomd/pytest/test_snapshot.py

```diff
@@ -309,3 +309,16 @@ def test_from_gsd_snapshot_populated(s, device):
     gsd_snap = make_gsd_snapshot(s)
     hoomd_snap = Snapshot.from_gsd_snapshot(gsd_snap, device.communicator)
     assert_equivalent_snapshots(gsd_snap, hoomd_snap)
+
+
+def test_invalid_particle_types(simulation_factory, lattice_snapshot_factory):
```

Comment:
I think we should emphasis the distinction between `types` and `typeids`.
```suggestion
def test_invalid_particle_typeids(simulation_factory, lattice_snapshot_factory):
```